### PR TITLE
Swallow Throwables thrown when configuring bugsnag appender

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Not all users use Logback in their project, so `BugsnagSpringConfiguration` calling `BugsnagAppender` methods can throw a `NoClassDefFoundError`, as observed in #132. If the dependency is not on the classpath, we should ignore this error and continue initialising bugsnag as normal.

## Tests
Ran the example app with [logback excluded](https://github.com/bugsnag/bugsnag-java/compare/log4j-example?expand=1), and confirmed that this change allowed the application to startup.

### CODE REVIEW

File: bugsnag-spring/src/main/java/com/bugsnag/BugsnagSpringConfiguration.java [hunk 1]
```diff
@@ -48,21 +48,25 @@ ScheduledTaskBeanLocator scheduledTaskBeanLocator() {
...
-        BugsnagAppender.addExcludedLoggerPattern("org.eclipse.jetty.server.HttpChannel");
+            // Exclude Jetty logger when processing HTTP requests via the HttpChannel
+            BugsnagAppender.addExcludedLoggerPattern("org.eclipse.jetty.server.HttpChannel");
 
-        // Exclude Undertow logger when processing HTTP requests
-        BugsnagAppender.addExcludedLoggerPattern("io.undertow.request");
+            // Exclude Undertow logger when processing HTTP requests
+            BugsnagAppender.addExcludedLoggerPattern("io.undertow.request");
+        } catch (Throwable ignored) {
+        }
```
Comment (kattrali) [inline bugsnag-spring/src/main/java/com/bugsnag/BugsnagSpringConfiguration.java]: 🚗 ))) Drive-by review: There should be a comment indicating why an exception is ignored here. Also, can the `Throwable` be a more specific subclass?

Comment (kinbiko) [reply bugsnag-spring/src/main/java/com/bugsnag/BugsnagSpringConfiguration.java]: I may be wrong, but after we do this I think we'd be able to remove the `@SuppressWarnings` annotation. I think a comment is sufficiently non-empty.
